packages/cli/src/lib/schemaHelpers.js

```diff
@@ -51,64 +51,59 @@ const getExistingModelName = async (name) => {
  * entire schema is returned.
  */
 export const getSchema = async (name) => {
-  if (name) {
-    const modelName = await getExistingModelName(name)
-    if (!modelName) {
-      throw new Error(
-        `No schema definition found for \`${name}\` in schema.prisma file`,
-      )
-    }
-    if (!schemaMemo[modelName]) {
-      const schema = await getSchemaDefinitions()
-      const model = schema.datamodel.models.find((model) => {
-        return model.name === modelName
-      })
-
-      if (model) {
-        // look for any fields that are enums and attach the possible enum values
-        // so we can put them in generated test files
-        model.fields.forEach((field) => {
-          const fieldEnum = schema.datamodel.enums.find((e) => {
-            return field.type === e.name
-          })
-          if (fieldEnum) {
-            field.enumValues = fieldEnum.values
-          }
-        })
-
-        // memoize based on the model name
-        schemaMemo[modelName] = model
-      }
-    }
+  const schema = (await getSchemaDefinitions()).datamodel
+  if (!name) {
+    return schema
+  }
+
+  const modelName = await getExistingModelName(name)
+  if (!modelName) {
+    throw new Error(
+      `No schema definition found for \`${name}\` in schema.prisma file`,
+    )
+  }
+
+  if (schemaMemo[modelName]) {
     return schemaMemo[modelName]
-  } else {
-    return (await getSchemaDefinitions()).datamodel
   }
+
+  const model = schema.models.find((model) => model.name === modelName)
+  if (!model) {
+    return undefined // can this happen, and if yes, should we prefer throwing an error?
+  }
+
+  // look for any fields that are enums and attach the possible enum values
+  // so we can put them in generated test files
+  model.fields.forEach((field) => {
+    const fieldEnum = schema.enums.find((e) => field.type === e.name)
+    if (fieldEnum) {
+      field.enumValues = fieldEnum.values
+    }
+  })
+
+  // memoize based on the model name
+  return (schemaMemo[modelName] = model)
```

Comment:
This is clever. A bit too clever for my taste :D

Reply:
Fair enough :laughing: